mono/metadata/class-internals.h

```diff
@@ -632,8 +626,9 @@ struct _MonoGenericContainer {
 
 #define mono_generic_param_owner(p)		((p)->owner)
 #define mono_generic_param_num(p)		((p)->num)
-#define mono_generic_param_info(p)		(mono_generic_param_owner (p) ? &((MonoGenericParamFull *) p)->info : NULL)
-#define mono_generic_param_name(p)		(((mono_generic_param_owner (p) || (p)->gshared_constraint)) ? ((MonoGenericParamFull *) p)->info.name : NULL)
+#define mono_generic_param_is_fullsize(p) ( !mono_generic_param_owner (p)->is_small_param )
```

Comment:
I'm with Zoltan on this one.

Those macros are now overly complicated and deserve to be turned into static inline functions.